File: CS/2108_Find_First_Palindromic_String_in_the_Array.cs
public class Solution {
    public string FirstPalindrome(string[] words) {
        string result = "";
        foreach(string word in words){
            if(isPalindromic(word)){
                result = word;
                break;
            }
        }
        return result;
    }
    private bool isPalindromic(string str){
        bool flag = true;
        for(int i = 0; i <= str.Length/2; i++){
            if(str[i] != str[str.Length-1 - i]){
                flag = false;
            }
        }
        return flag;
    }
}

File: Cpp/2108_Find_First_Palindromic_String_in_the_Array.cpp
class Solution {
public:
    string firstPalindrome(vector<string>& words) {
        for(string word : words){
            if(isPalindromic(word)){
                return word;
            }
        }
        return "";
    }
private:
    bool isPalindromic(string s){
        int n = s.size();
        for(int i = 0; i < n/2; i++){
            if(s[i] != s[n-1-i]){
                return false;
            }
        }
        return true;
    }
};
